linkerd/tls/src/client.rs

```diff
@@ -58,9 +58,18 @@ pub struct Client<L, C> {
 
 #[pin_project::pin_project(project = ConnectProj)]
 #[derive(Debug)]
-pub enum Connect<F, I, H: Service<I>> {
-    Connect(#[pin] F, Option<H>),
-    Handshake(#[pin] Oneshot<H, I>),
+pub enum Connect<F, I, H: Service<I>, M> {
+    Connect(#[pin] F, Option<Conditional<H, NoClientTls>>),
+    Handshake(
+        #[pin] Oneshot<H, I>,
+        Option<(Conditional<(), NoClientTls>, M)>,
+    ),
```

Comment:
nit, take it or leave it: i might give these named fields? not a blocker though.